engine/src/main/java/org/terasology/config/flexible/ui/NumberRangeConstraintWidgetFactory.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible.ui;
+
+import org.terasology.config.flexible.constraints.NumberRangeConstraint;
+import org.terasology.rendering.nui.UIWidget;
+import org.terasology.rendering.nui.databinding.Binding;
+import org.terasology.rendering.nui.widgets.UISlider;
+
+public class NumberRangeConstraintWidgetFactory<T extends Number & Comparable<? super T>>
+    extends AssetBackedConstraintWidgetFactory<T, NumberRangeConstraint<T>> {
+
+    public NumberRangeConstraintWidgetFactory() {
+        super("engine:numberRangeSettingWidget");
+    }
+
+    @Override
+    protected void bindWidgetToSetting(UIWidget widget) {
+        UISlider slider = widget.find("slider", UISlider.class);
+        assert slider != null;
+
+        updateSliderIfInteger(slider);
+
+        setSliderRange(slider);
+
+        slider.bindValue(new Binding<Float>() {
+            @Override
+            public Float get() {
+                return getSetting().getValue().floatValue();
+            }
+
+            @Override
+            public void set(Float value) {
+                setSettingValue(value);
+            }
+        });
+    }
+
+    private void updateSliderIfInteger(UISlider slider) {
+        T settingValue = getSetting().getValue();
+
+        if (settingValue instanceof Float || settingValue instanceof Double) {
+            return;
+        }
+
+        slider.setIncrement(1.0f);
```

Comment:
Doesn't this also trigger on `Byte`, `Short` and `Long` rather than `Integer`?

Reply:
Yes, this code intends to set the increment to `1.0` for all non-decimal numeric types.